### PR TITLE
Week 4 APIs(Maps/Charts) - Uday

### PR DESCRIPTION
**Pull request by UdayKalra**
Implemented Maps API to show university on contact page and Charts API to endorse skills on resume page.

### CODE REVIEW

File: udaykalra/portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -124,7 +139,6 @@ div.main-content {
   background-color: var(--body-color);
   color: var(--main-color);
   font: 'Roboto', sans-serif;
-
   flex: 1;
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/style.css:142]: Sort alphabetically?

Also the html, body rules below.

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -20,13 +20,55 @@ async function deleteCommentData() {
...
+}
+
+google.charts.load('current', {'packages': ['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+/** Fetches skill data and uses it to create a chart. */
+function drawChart() {
+  fetch('/endorse-data')
+      .then(response => response.json())
+      .then((skillVotes) => {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:51]: I'm having a hard time figuring out what kind of values `skillVotes` contains.

It'd be helpful for me if there were comments about what it contains.

My guess is that this is an array of objects having a key == skill and object == number of votes, but I'm not sure.

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/PollDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/endorse-data")
+public class PollDataServlet extends HttpServlet {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/PollDataServlet.java:28]: Style requires documentation for public classes:
https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

Documentation would also be useful to the reader to understand that "Poll" means "Endorsed Skills".

Optional: renaming this to EndorsedSkillsDataServlet may be clearer.

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -20,13 +20,55 @@ async function deleteCommentData() {
...
+        const data = new google.visualization.DataTable();
+        data.addColumn('string', 'Skill');
+        data.addColumn('number', 'Votes');
+        Object.keys(skillVotes).forEach((skill) => {
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:55]: `Object.keys(skillVotes).forEach((skillVote)`

Naming the singular form after the plural form is less confusing (when they are different I wonder whether some other transformation was made).

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -20,13 +20,55 @@ async function deleteCommentData() {
...
 }
+
 /** Creates an <li> element containing text. */
 function createCommentListItem(inputText) {
   const listElement = document.createElement('li');
   listElement.innerText = inputText;
   return listElement;
 }
 
+/** Creates a map to show University. */
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:31]: /** Creates a map with a marker at UC San Diego. */

Would be more specific.

File: udaykalra/portfolio/src/main/webapp/style.css (hunk 1 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/style.css:141]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/PollDataServlet.java (hunk 3 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/PollDataServlet.java]: Done

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/script.js]: Done

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/script.js]: Done

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/script.js:51]: Done

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/EndorsedDataServlet.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/endorse-data")
+public class EndorsedDataServlet extends HttpServlet {
+  // Store skills and corresponding votes in a HashMap/
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/java/com/google/sps/servlets/EndorsedDataServlet.java:29]: The trailing slash seems to be a typo.

File: udaykalra/portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -20,13 +20,55 @@ async function deleteCommentData() {
...
   return listElement;
 }
 
+/** Creates a map with marker at UC San Diego */
+function createMap() {
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 32.879838, lng: -117.232351}, zoom: 16});
+
+  const trexMarker = new google.maps.Marker({
```
Comment (ykjchen) [inline udaykalra/portfolio/src/main/webapp/script.js:37]: Out of curiosity, what does trex mean here?

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (ykjchen) [reply udaykalra/portfolio/src/main/webapp/script.js:51]: Maybe add here:

```
The skillVotes variable is an array of objects that each have a single key-value pair where the key is a string representing a skill and the value is a number representing the number of endorsements received for that skill.
```

File: udaykalra/portfolio/src/main/java/com/google/sps/servlets/EndorsedDataServlet.java (hunk 6 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/java/com/google/sps/servlets/EndorsedDataServlet.java]: Done

File: udaykalra/portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (UdayKalra) [reply udaykalra/portfolio/src/main/webapp/script.js]: Changed to schoolMarker, original name was temporary.

**Pull request merged**